### DISCUSSION TITLE
Custom icons in badges

### DISCUSSION THREAD

**Discussion by IndianBartonka** [badges/shields#8872] [Ideas] [answered]
I can use custom icon on badge?

**Comment by IndianBartonka** [comment]
![1675329166158](https://user-images.githubusercontent.com/68228599/216283776-72e99050-860a-43ae-8650-24cf18c0e4b2.png)

**Comment by chris48s** [comment, marked as answer]
You can pass a base64 encoded custom logo as a query parameter. There are docs on this at https://github.com/badges/shields/blob/master/doc/logos.md#custom-logos However, there is a bit of a constraint on this: There is a limit on the total size of request headers we can accept (8192 bytes). This means the base64-encoded image text is limited to somewhere slightly under 8192 bytes depending on the rest of the request header.

In practice, this usually means that your custom logo needs to be a SVG. We theoretically can accept a PNG in a base64 string but most real-world PNGs are too big and hit the size limit.

**Comment by IndianBartonka** [comment]
Thx
